### DISCUSSION TITLE
Why we need composables?

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#18032] [Questions] [answered]
Please tell me why I need to use `composables`?

**Comment by geetfun** [comment, marked as answer]
> Please tell me why I need to use `composables`?
> 
> 

From personal experience, depending on the size and logic of the component, a composable is not needed. However, if there's a lot of logic there and it appears that the logic is re-used elsewhere, then a composable is a pattern where you can extract that logic out.

See: https://vueschool.io/articles/vuejs-tutorials/what-is-a-vue-js-composable/

```
const { profilePicture, profileName } = useProfile();
```

The above, for example, could be called in multiple components and may have the same logic. So instead of redeclaring it separately, the composable serves to consolidate things a bit.
